src/test/scala/ModeTest.scala

```diff
@@ -0,0 +1,42 @@
+package chess
+
+import chess.{ ChessTest, Mode }
+import chess.Mode._
+
+class ModeTest extends ChessTest {
+
+  private val invalidId = 42
+
+  "Mode" should {
+    "either be Casual or Rated" in {
+      Casual.casual must_== true
+      Casual.rated must_== false
```

Comment:
I'm sorry but I don't understand the point of these tests, and other similar tests. How does that just not repeat the equally simple code:

```
  def casual = this == Mode.Casual
  def rated  = this == Mode.Rated
```

I think tests for the sake of testing is not a good thing.